### PR TITLE
decrease button size, increase icon size

### PR DESCRIPTION
**Pull request by murggel**
## Description

* change button size and icon size:
button size: 42dp
icon size: 30dp
* adapt height of "Details"-button in cache-popup
* align routing-buttons in map-settings

## Related issues

rel #11065

## Additional context

screenshots:
https://github.com/cgeo/cgeo/issues/11065#issuecomment-873455080
https://github.com/cgeo/cgeo/issues/11065#issuecomment-873455127
https://github.com/cgeo/cgeo/issues/11065#issuecomment-873455684
https://github.com/cgeo/cgeo/issues/11065#issuecomment-873455309

### CODE REVIEW

**Comment by murggel** [comment]
Have to check it for API 21

**Comment by murggel** [comment]
ready for review

File: main/res/values/dimens.xml [hunk 1]
```diff
@@ -42,7 +42,10 @@
 
     <!-- buttons -->
     <dimen name="textSize_buttonsPrimary">@dimen/textSize_detailsSecondary</dimen>
-w
+
+    <!-- icons -->
+    <dimen name="icon_size">42dp</dimen>
```
Comment (moving-bits) [inline main/res/values/dimens.xml:47]: just a minor thing: the icon itself has a size of `30dp` (as defined in the style below), `42dp` is for the whole button, so maybe it should be called something like `iconButtonSize`, and the setting could be in the same group as `textSize_buttonsPrimary`.

**Review by moving-bits — APPROVED** [review]
LGTM, thanks @murggel for tackling the icon button issue!

**Review by eddiemuc — APPROVED** [review]
LGTM

**Comment by Lineflyer** [comment]
Approved by two members. Merging.

File: main/res/values/dimens.xml (hunk 1 above)
Comment (murggel) [reply main/res/values/dimens.xml:47]: Will fix that in one of the next PRs...

**Comment by fm-sys** [comment]
Sorry, I missed the discussion ccompletely but I'm convinced that the button size is now way too small! 

The larger button size was one of the main improvements which came along with the new material theme. And while using the latest nightly today while my phone was in an waterproof case I really had trouble to get the correct button pressed. Sometimes, it took five times until the button did react. While it was no problem at all with the previous versions... 

We simply cannot completly ignore usability, just because we want the icon buttons in the same size as text buttons. 

Screenshot_20210705_203427_org mozilla firefox

**Comment by murggel** [comment]
I read something about sizes from 42dp and higher.
But of course we can change back to 48dp and e.g. 32dp for the icon.

It is a attribte in dimens.xml, so easy to change.

**Comment by moving-bits** [comment]
Never had any problems with the buttons we have in our current release, and those are much smaller as those we have in current master. The "old new" buttons had been too large compared to the rest of the page, IMHO (but, agreed, that's also a matter of taste). I'll do a test on a physical device, just started charging an old mobile :-)

IIRC 48dp does not necessarily mean the button (or touch target) itself, but includes the margins. So a button with 42dp and 3dp margin on either side should be fine.

**Comment by Lineflyer** [comment]
I also think the current size is OK: Larger than the "legacy" ones and easily clickable. The shouldn't be smaller though...so we used kind of the minimum. Not having measured it, I guess we are at about 7mm for the smallest?

**Comment by murggel** [comment]
The size of the action-bar is 45dp, shall we use that size?

**Comment by fm-sys** [comment]
> The size of the action-bar is 45dip, shall we use that size?

I guess, that would be a good compromise. With the "old new" we where at about 9mm and with the "new New" we are exactly at almost 7mm (have just messured). So 8mm (45dp) sounds like a good compromise ;-)

**Comment by fm-sys** [comment]
> dip and dp is the same, correct?

Yes. At least AFAIK...

**Comment by moving-bits** [comment]
To cite another issue's entry: We are supposed to [haggle](https://www.youtube.com/watch?v=-2iZjxSGca8)... ;-)

**Comment by moving-bits** [comment]
> dip and dp is the same, correct?

Yes, `dip` seems to be the older name for it, current documentation uses `dp` (AFAICS).

This article has a nice overview: https://medium.com/analytics-vidhya/what-is-the-difference-between-px-dip-dp-and-sp-e4351fefa685

**Comment by Lineflyer** [comment]
Interesting BTW: Learned from the article , that we should use "sp" to define font sizes to allow users to modify them as per their system font settings. Not sure if we do that at all places where it would be meaningful (e.g. cache description).

**Comment by murggel** [comment]
> So a button with 42dp and 3dp margin on either side should be fine.

That's exactly the "new" button size 🤔

**Comment by fm-sys** [comment]
> > So a button with 42dp and 3dp margin on either side should be fine.
> 
> That's exactly the "new" button size 🤔

Problem is, that the margin isn't part of the touch target... :-/

**Comment by moving-bits** [comment]
> Interesting BTW: Learned from the article , that we should use "sp" to define font sizes to allow users to modify them as per their system font settings. Not sure if we do that at all places where it would be meaningful (e.g. cache description).

This is our font size definition since our UI overhaul, defined in `res/values/dimens.xml`:

```
<!-- font size dimensions ====================================== -->

    <!-- all toolbars except main screen -->
    <dimen name="textSize_toolbarSingle">20sp</dimen>
    <dimen name="textSize_toolbarPrimary">18sp</dimen>
    <dimen name="textSize_toolbarSecondary">16sp</dimen>

    <!-- headings (not toolbar) -->
    <dimen name="textSize_headingPrimary">20sp</dimen>
    <dimen name="textSize_headingSecondary">18sp</dimen>

    <!-- lists -->
    <dimen name="textSize_listsPrimary">20sp</dimen>
    <dimen name="textSize_listsSecondary">16sp</dimen>

    <!-- calculator labels between input buttons -->
    <dimen name="textSize_calculator">22sp</dimen>

    <!--- wizard -->
    <dimen name="textSize_headingWizard">26sp</dimen>
    <dimen name="textSize_detailsWizard">20sp</dimen>

    <!-- buttons -->
    <dimen name="textSize_buttonsPrimary">@dimen/textSize_detailsSecondary</dimen>

    <!-- all other text -->
    <dimen name="textSize_detailsPrimary">16sp</dimen>
    <dimen name="textSize_detailsSecondary">14sp</dimen>
```

**Comment by fm-sys** [comment]
BTW, just discovered a "missing single-line parameter" bug which we should fix on the go...

grafik

**Comment by Lineflyer** [comment]
@moving-bits Looks perfect....except...do we want button text to resize if the user selects larger fonts....have to try it out next days how this will look like

**Comment by fm-sys** [comment]
I've just created PR #11107. If the beta is not yet compiling, this should be included as well. Especially because of some ugly quirks in the current build...

Screenshot_20210705_220727_cgeo geocaching_copy_389x222

**Pull request merged**